mlflow/utils/search_utils.py

```diff
@@ -1846,3 +1814,100 @@ def _get_comparison(cls, comparison):
         comp["comparator"] = stripped_comparison[1].value
         comp["value"] = cls._get_value(comp.get("type"), comp.get("key"), stripped_comparison[2])
         return comp
+
+
+class SearchLoggedModelsUtils(SearchUtils):
+    NUMERIC_ATTRIBUTES = {
+        "creation_timestamp",
+        "creation_time",
+        "last_updated_timestamp",
+        "last_updated_time",
+    }
+    VALID_SEARCH_ATTRIBUTE_KEYS = {
+        "name",
+        "model_id",
+        "model_type",
+        "status",
+        "source_run_id",
+    } | NUMERIC_ATTRIBUTES
+    VALID_ORDER_BY_ATTRIBUTE_KEYS = VALID_SEARCH_ATTRIBUTE_KEYS
+
+    @classmethod
+    def _does_logged_model_match_clause(cls, model: LoggedModel, sed: dict[str, Any]):
+        key_type = sed.get("type")
```

Comment:
potentially annoying nit: can we use another name here that doesn't mirror a linux command line tool? First thing that popped in my head when using this was ... "wait, what? huh? ohhhhh"